src/app/services/tree-node.service.ts

```diff
@@ -257,6 +257,9 @@ export class TreeNodeService {
       concept.code = treeNode.conceptCode;
       concept.fullName = treeNode.fullName;
       concept.name = treeNode.name;
+      if (treeNode.metadata && treeNode.metadata['subject_dimension']) {
```

Comment:
I was wondering why we would restrict to a single subject dimension, but since the metadata is a simple key-value map, this is a good approach. If we want to support multiple dimensions later, we could parse a `subject_dimensions` metadata element and choose a proper separator character then.